tests/launcher/test_local.py

```diff
@@ -187,3 +191,75 @@ def func_b():
     # rerunning temp_b with temp_a dependence and verifying with the updated temp_a code
     run = project.run_function("func", local=True, reset_on_run=True)
     assert run.output("return") == "dummy value updated"
+
+
+@pytest.mark.parametrize(
+    ["batching", "batch_size"], [(False, None), (True, None), (True, 10)]
+)
+def test_run_local_serving_job(batching, batch_size):
+    project = mlrun.new_project("some-project")
+    function = mlrun.code_to_function(
+        name="test", kind="serving", filename=str(custom_classes_path)
+    )
+    graph = function.set_topology("flow", engine="async")
+    graph.to(name="increaser", class_name="SepalLengthIncreaser").respond()
+    job = function.to_job()
+
+    inputs = {"data": str(input_csv_path)}
+    params = {"batching": batching, "batch_size": batch_size}
+
+    result = project.run_function(job, inputs=inputs, params=params, local=True)
+    responses, termination_result = eval(result.status.results["return"])
+
+    num_input_rows = 150  # number of rows in input file
+    if batching:
+        num_expected_responses = (
+            math.ceil(num_input_rows / batch_size) if batch_size else 1
+        )
+    else:
+        num_expected_responses = 150
+    assert len(responses) == num_expected_responses
+
+    first_response = responses[0]
+    if batching:
+        first_response = first_response[0]
+    assert first_response == {  # based on the first row in input file
+        "sepal_length": 6.1,
+        "sepal_width": 3.5,
+        "petal_length": 1.4,
+        "petal_width": 0.2,
+        "species": "setosa",
+    }
+
+    assert termination_result is None
+
+
+@pytest.mark.parametrize("with_target_mapping", [False, True])
+def test_run_local_serving_job_with_target(with_target_mapping):
+    project = mlrun.new_project("some-project")
+    function = mlrun.code_to_function(
+        name="test", kind="serving", filename=str(custom_classes_path)
+    )
+    graph = function.set_topology("flow", engine="async")
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        graph.to(name="increaser", class_name="SepalLengthIncreaser")
+        graph.to(name="parquet", class_name="storey.ParquetTarget", path=tmp_dir)
+
+        target_mapping = None
+        if with_target_mapping:
+            new_target = params_to_step(
```

Comment:
This is probably not how users will actually do this - I would expect them to use the MLRun classes, such as `mlrun.datastore.ParquetTarget` directly, rather than use this internal `params_to_step` function.

Reply:
Then they'll need to pass this as a string, same as we do everywhere else (i.e. in `step.to()`). That is to say, we can call `params_to_step` internally for the user.

Reply:
This really goes back to the question of how target steps are represented in the graph today. The users can create class instances of those targets, or we can indeed call `params_to_step` for them. My concern is mostly that `params_to_step` is not widely documented and is mostly an internal function, so it's better if we do this for the users like we do in other places.